src/Illuminate/Database/Eloquent/Relations/Concerns/CanBeOneOfMany.php

```diff
@@ -131,8 +131,6 @@ public function ofMany($column = 'id', $aggregate = 'MAX', $relation = null)
             ];
         }
 
-        $this->addConstraints();
```

Comment:
Why is this removed?

Reply:
The removal of this constraint causes all rows in the database to be loaded and than filtered while before the query was more narrow and only returned the required results. I have opened an [issue](https://github.com/laravel/framework/issues/55155) for this, I can create a PR reverting this change but I don't know what the impact is on the feature that was added in this PR.

Reply:
@BertvanHoekelen, the constraints should have already been added by the time this was called---see the tests, this was adding double constraints.

Can you show the code used (here and on your issue) that resulted in the query?